### DISCUSSION TITLE
Getting error after scanning the QR with Binance Defi wallet and facing chainId issue with Metamask wallet

### DISCUSSION THREAD

**Discussion by mohit-zone** [WalletConnect/walletconnect-monorepo#2767] [Developer Support]
I have used the @walletconnect/ethereum-provider to migrate from v1 to v2 .
Repository Referrence:-https://github.com/mohitbadgotra/walletconnectv2/blob/main/app/page.tsx

const client = await EthereumProvider.init({
        projectId: process.env.NEXT_PUBLIC_PROJECT_ID,
        showQrModal: true,
        chains: [97],//want to connect BSC Chain
        methods: ["eth_sendTransaction", "personal_sign"],
        events: ["chainChanged", "accountsChanged"],
      });
      await client.connect();

On Scanning QR with Binance Defi wallet i am getting error call:fail Wallet connect uri unavailable
![Binance Defi Wallet](https://github.com/WalletConnect/walletconnect-monorepo/assets/128027452/0af33bfb-4ad7-4f0c-b568-7cb684b57a4f)

On Scanning the QR with Metamask wallet I am able to connect but getting incorrect connected wallet chain id.Like on Metamask wallet i have selected Ethereum Mainnet network but even then when i am fetching chain id on web it gives me chain id of BSC Testnet.
 const web3 = new Web3(client);
      const connectedChainId = await web3.eth.getChainId();
      console.log("connectedChainId", connectedChainId);
      
      Ideally connectedChainId should be 1 but its giving 97 as response.How can i fetch the chain id of network which is selected in metamask wallet.
      
![Screenshot_2023-06-26-13-15-02-133_io metamask](https://github.com/WalletConnect/walletconnect-monorepo/assets/128027452/72e757ea-e0da-454a-9551-9300338f978a)

**Comment by finessevanes** [comment]
Two things:
1. can you provide the URI you're scanning? Are you certain the wallet you're using supports v2? To test v2 support, try connecting to our [example dapp](https://react-app.walletconnect.com/)
2. you're not getting the chain id because in your set up `chains: [97],//want to connect BSC Chain` you only have BSC. You will need to add eth mainnet to that array

**Comment by mohit-zone** [comment]
1.Got same error on example dapp.I am using Binance andriod app and it was working with walletconnect v1 but with v2 its not working .
2.On adding the mainnet chain id to array ,got unsupported chain id error
<img width="1423" alt="Screenshot 2023-06-26 at 7 00 25 PM" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/128027452/2eed4d3e-6760-46ef-a9ed-0e7d7e7a538e">

**Comment by tiennt0212** [comment]
I face the same error on chainId before.

I'm working on the Front-end side with ReactJS

I use a variable and update its value every time I receive a new chainId.

In truth, I use a state value, which is a concept in Web Front-end developers. Therefore, every time the state's value was updated, the user interface will renew.

Just my way to adapt it :sweat_smile:

**Reply by mohit-zone** [reply]
thanks @tiennt0212 but i have handled the case by adding delay by using setTimeout to 
const connectedChainId = await web3.eth.getChainId();
It worked for metamask this way but for binance defi wallet i am still getting error call:fail wallet connect uri unavailable
